plugins/modules/clustering/consul/consul_kv.py

```diff
@@ -247,6 +248,7 @@ def set_value(module):
     if changed and not module.check_mode:
         changed = consul_api.kv.put(key, value,
                                     cas=module.params.get('cas'),
+                                    acquire=session,
```

Comment:
I'm not sure, but this change might be wrong. The documentation says that `session` is used when "acquir[ing] or releas[ing] a lock", which is done when `state=acquire` or `state=release`. This code is called when `state=present`. The main difference between `acquire` and `present` seems to be whether `acquire=session` is passed or not.

Reply:
indeed, using `state=acquire` is enough. This can be tested like so:
```yaml
---
- name: test
  hosts: localhost
  gather_facts: false
  tasks:
    - name: Create ephemeral session
      community.general.consul_session:
        name: "ephemeral-session"
        ttl: 60
        state: present
        behavior: delete
      register: ephemeral_session

    - name: Create ephemeral key
      community.general.consul_kv:
        key: "ephemeral-key"
        value: 1234
        session: "{{ ephemeral_session.session_id }}"
        state: acquire
```
You can start consul like this
```
docker run -it -p 8500:8500 consul
```
As such I've removed the second commit.